fix: catch reinit of replicaset

boredland opened the pull request:
- #278

hasezoey commented on .gitignore:
```diff
@@ -46,3 +46,6 @@ packages/mongodb-memory-server-core/lib
 package-lock.json
+
+# testfiles
+reuse*
```
why do these need to be ignored?

hasezoey commented on packages/mongodb-memory-server-core/src/__tests__/replset-single-restart-test.ts:
```diff
@@ -0,0 +1,32 @@
+
+describe('multi-member replica set', () => {
+  it('should start multiple times', async () => {
+    const testDir = fs.mkdtempSync('reuse_');
```
please refactor to use tmp (package)

hasezoey commented on yarn.lock:
```diff
@@ -1839,6 +1839,14 @@
   resolved "https://registry.yarnpkg.com/@types/node/-/node-12.12.21.tgz#aa44a6363291c7037111c47e4661ad210aded23f"
   integrity sha512-8sRGhbpU+ck1n0PGAUgVrWrWdjSW2aqNeyC15W88GRsMpSwzv6RJGlLhE7s2RhVSOdyDmxbqlWSeThq4/7xqlA==
 
+"@types/rimraf@^2.0.3":
```
i think this should be removed here too

hasezoey commented:
are you sure you reverted the lockfile correctly?

boredland commented:
> are you sure you reverted the lockfile correctly?
> -> seems like more is now removed than earlier commits added

I"m sure I reverted that one correctly, but now I just squashed those changes together to cover the unnecessary changes.

hasezoey commented:
@boredland checked it out locally, and the yarn.lock is still not like the master one (many things removed and set to old versions)

to fix this run `yarn install` in the root of the repo or in the directory the yarn.lock file is in and commit it, you will see if you compare that with nodkz/master that only 3 package versions have changed
-> i would have pushed it to your branch, but it seems like "Allow edits from maintainers" is disabled for this pr / is not possible when you dont own the repo/branch

hasezoey commented on packages/mongodb-memory-server-core/src/__tests__/replset-single-restart-test.ts:
```diff
@@ -0,0 +1,47 @@
+};
+
+describe('single-member replica set', () => {
+  it('should start multiple times', async () => {
```
maybe an special test-timeout should be set for this test

hasezoey commented on packages/mongodb-memory-server-core/src/__tests__/replset-single-restart-test.ts:
```diff
@@ -0,0 +1,47 @@
+import MongoMemoryReplSet from '../MongoMemoryReplSet';
+import * as tmp from 'tmp';
+
+jasmine.DEFAULT_TIMEOUT_INTERVAL = 600000;
```
please remove this, otherwise it might affect other tests too

sakulstra commented:
hayho, what's the status here?

@hasezoey can we somehow help pushing this forward? :pray:

nodkz commented:
Everything is ok! Bump it!

Tnx 🙏

Ranguna commented:
Has this been released already ?
I can't seem to find these changes on my locally installed `mongodb-memory-server-core`.

hasezoey commented:
@Ranguna yes since 6.4.0

Ranguna commented:
@hasezoey all right, thanks!
